backend/inspirehep/records/api/institutions.py

```diff
@@ -16,3 +17,18 @@ class InstitutionsRecord(InspireRecord):
     es_serializer = InstitutionsElasticSearchSchema
     pid_type = "ins"
     pidstore_handler = PidStoreInstitutions
+
+    def delete_relations_with_literature(self):
+        InstitutionLiterature.query.filter_by(institution_uuid=self.id).delete()
+
+    def delete(self):
+        super().delete()
+        self.delete_relations_with_literature()
+
+    def hard_delete(self):
+        self.delete_relations_with_literature()
```

Comment:
AFAIS in the mixin you have already override the ``delete`` and ``hard_delete`` to delete the relations

Reply:
The mixin is on ```LiteratureRecord```. I need to delete the relations also on ```InstitutionRecord``` delete.